.github/workflows/release.yaml

```diff
@@ -23,31 +23,60 @@ on:
   push:
     branches:
       - main
+
+defaults:
+  run:
+    shell: nix develop --impure --command bash {0}
+
 jobs:
   release:
     runs-on: ubuntu-latest
     permissions:
       contents: write
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4 # v3.5.3
+      - uses: cachix/install-nix-action@v26
         with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      # Cache the nix store.
+      - uses: DeterminateSystems/magic-nix-cache-action@v4
+
+      # Try to checkout a branch with the same name as provided in the dispatch
+      - uses: actions/checkout@v4 # v3.5.3
+        id: checkout
+        with:
+          ref: ${{ github.event.client_payload.branch }}
```

Comment:
Can you provide an example of what the branch name would be here? Are we expecting this commit to go to `main`?

Reply:
Because we will always run on main it will always be `main`

https://github.com/redpanda-data/helm-charts/blob/b14967ada018fc9be743665040a6191f187ed65a/.github/workflows/release.yaml#L22-L25

I took it from one of our internal repository example.

I can change it to hardcoded `main`.

Reply:
AH, okay. Hard coded or just a quick comment at the end saying something like `branch will always be a branch from the list on line 24`